src/Cli/dotnet/CommandFactory/CommandResolution/MuxerCommandSpecMaker.cs

```diff
@@ -20,12 +20,22 @@ internal static class MuxerCommandSpecMaker
             {
                 throw new Exception(LocalizableStrings.UnableToLocateDotnetMultiplexer);
             }
+            IEnumerable<string> modifiedArguments = commandArguments;
+
+            // Add --roll-forward argument first if exists
+            if (commandArguments != null && commandArguments.Any(arg => arg.Equals("--roll-forward", StringComparison.OrdinalIgnoreCase)))
+            {
+                int index = commandArguments.ToList().IndexOf("--roll-forward");
+                arguments.Add(commandArguments.ElementAt(index));
+                arguments.Add(commandArguments.ElementAt(index + 1));
+                modifiedArguments = commandArguments.Where((element, i) => i != index && i != index + 1);
+            }
 
             arguments.Add(commandPath);
 
-            if (commandArguments != null)
+            if (modifiedArguments != null)
             {
-                arguments.AddRange(commandArguments);
+                arguments.AddRange(modifiedArguments);
```

Comment:
GitHub doesn't let me comment on the entire method, but this is how I'd write it. It isn't necessary to do all the manual reshuffling of stuff. The only thing you need to know is if the current or previous argument is `--roll-forward`. So, I just grouped them by that predicate and then create the result set with the values in the grouped subsets.

```C#
        internal static CommandSpec CreatePackageCommandSpecUsingMuxer(
            string commandPath,
            IEnumerable<string> commandArguments)
        {
            var muxer = new Muxer();
            var host = muxer.MuxerPath;
            if (host == null)
            {
                throw new Exception(LocalizableStrings.UnableToLocateDotnetMultiplexer);
            }
            string previousArg = string.Empty;
            // Group the arguments by if the previous argument or the current argument is --roll-forward.
            // The use of bitwise OR is necessary here as we don't want to short-circuit the boolean calculation since previousArg always needs to be reassigned.
            var argGroups = (commandArguments ?? [])
                .GroupBy(a => previousArg.Equals("--roll-forward", StringComparison.OrdinalIgnoreCase) | (previousArg = a).Equals("--roll-forward", StringComparison.OrdinalIgnoreCase))
                .ToArray();

            string[] arguments = [.. argGroups.Where(g => g.Key).SelectMany(g => g), commandPath, .. argGroups.Where(g => !g.Key).SelectMany(g => g)];
            return CreateCommandSpec(host, arguments);
        }
```

Here's the test console app I wrote for it:
```C#
string commandPath = "whatever";
IEnumerable<string> commandArguments = new List<string> { "a", "b", "--roll-forward", "value", "c", "d", "e" };

string previousArg = string.Empty;
// Group the arguments by if the previous argument or the current argument is --roll-forward.
// The use of bitwise OR is necessary here as we don't want to short-circuit the boolean calculation since previousArg always needs to be reassigned.
var argGroups = (commandArguments ?? [])
    .GroupBy(a => previousArg.Equals("--roll-forward", StringComparison.OrdinalIgnoreCase) | (previousArg = a).Equals("--roll-forward", StringComparison.OrdinalIgnoreCase))
    .ToArray();

List<string> result = [..argGroups.Where(g => g.Key).SelectMany(g => g), commandPath, ..argGroups.Where(g => !g.Key).SelectMany(g => g)];

result.ForEach(Console.WriteLine);
```

The output of the console app looks like this:
```
--roll-forward
value
whatever
a
b
c
d
e
```

Reply:
Thank you so much for the comment!